builtin/plugin/backend.go

```diff
@@ -49,23 +49,34 @@ func Backend(ctx context.Context, conf *logical.BackendConfig) (logical.Backend,
 
 	sys := conf.System
 
-	// NewBackend with isMetadataMode set to true
-	raw, err := bplugin.NewBackend(ctx, name, pluginType, sys, conf, true)
+	// NewBackend with isMetadataMode set to false
+	raw, err := bplugin.NewBackend(ctx, name, pluginType, sys, conf, false)
 	if err != nil {
 		return nil, err
 	}
-	err = raw.Setup(ctx, conf)
-	if err != nil {
-		raw.Cleanup(ctx)
-		return nil, err
+
+	// HACK: Cast to BackendPluginClient
```

Comment:
Nit: Suggest to change the comment to explain what's not obvious from reading the code. For example, it might be good to explain why this is needed, why it's classified as a hack, or other subtle bits around the behavior that would be important to remember in the future.